pkg/util/template/effective_resource.go

```diff
@@ -0,0 +1,82 @@
+package template
+
+import (
+	"fmt"
+	htmltemplate "html/template"
+	texttemplate "text/template"
+)
+
+type EffectiveResourceView interface {
+	fn()
+}
+
+type EffectiveResource struct {
+}
+
+func (v EffectiveResource) fn() {}
+
+var _ EffectiveResourceView = &EffectiveResource{}
+
+// An intermediate view to be resolved to html and text view
+type TemplatesEffectiveResourceIntermediateView interface {
+}
+
+type TemplatesEffectiveIntermediateResource struct {
+	Data        []byte
+	LanguageTag languageTag
+}
+
+var _ TemplatesEffectiveResourceIntermediateView = &TemplatesEffectiveIntermediateResource{}
+
+type HTMLTemplatesEffectiveResourceView interface {
+	EffectiveResourceView
+	TemplatesEffectiveResourceIntermediateView
+}
+
+type HTMLTemplatesEffectiveResource struct {
+	EffectiveResource
+	*TemplatesEffectiveIntermediateResource
+	Template *htmltemplate.Template
+}
```

Comment:
Simplify this struct to 

```
type HTMLTemplateEffectiveResource struct {
  Data []byte
  LanguageTag String
  Template *htmltemplate.Template
}
```

Note that **there is no s** in `Template`

Apply this to TextTemplateEffectiveResource

Note that **there is no s** in `Template`